### PR TITLE
fix: Resolved bug with extract func in lua with assigned functions

### PR DESCRIPTION
**Pull request by teddylear**
fix: Resolved bug with extract func in lua with assigned functions

Resolves parts of (if not all of) #292

### CODE REVIEW

File: lua/refactoring/tests/refactor/106/lua/assign-func/extract.start.lua [hunk 1]
```diff
@@ -0,0 +1,11 @@
+-- stylua: ignore start
```
Comment (pranavrao145) [inline lua/refactoring/tests/refactor/106/lua/assign-func/extract.start.lua:1]: Technically we don't really need this anymore because I added a .styluaignore a bit ago, but it's fine - I'm going to make another PR soon anyway to get rid of these at the top of all the lua tests.

**Review by pranavrao145 — APPROVED** [review]
LGTM!

File: lua/refactoring/tests/refactor/106/lua/assign-func/extract.start.lua (hunk 1 above)
Comment (teddylear) [reply lua/refactoring/tests/refactor/106/lua/assign-func/extract.start.lua:1]: Oh didn't know you added that, nice!
